sdk/spring/pom.xml

```diff
@@ -6,8 +6,80 @@
   <artifactId>spring-cloud-azure</artifactId>
   <name>Spring Cloud Azure Aggregator</name>
   <packaging>pom</packaging>
-  <version>1.0.0</version><!-- Need not change for every release-->
+  <version>4.1.0-beta.1</version> <!-- {x-version-update;com.azure.spring:spring-cloud-azure;current} -->
```

Comment:
@stliu 
This is not something that can be checked in for the following reasons:
1. sdk/spring/pom.xml is part of our build hierarchy these are not product or releasable files.
2. The pipelines for SDK/<areas> only handle libraries, not libraries and POMs. The only pipeline right now that handles POM only releases is the BOM pipeline. You wouldn't see this until release time but a POM only release would completely fail.
3. Starter libraries aren't azure-client-sdk libraries in the strictest sense, they don't inherit from azure-client-sdk-parent (which you wouldn't want because it would drag a number of build rules/targets amongst other things) which is where you'd normally put these type of items.

I realize that there's a bunch of duplication in the starters' pom files but these aren't items that are changing with any kind of frequency, if at all. Even ignoring the release problems, the only thing creating parent file does is create another artifact that needs to be released but isn't actually necessary.

Reply:
@JimSuplizio the intention of this pr is making the pom as a parent pom for all spring modules, though in this pr, I only convert starters pom only, the rest of spring poms are going to change once this pr merged.

currently all spring poms are using this https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/parents/azure-client-sdk-parent/pom.xml as parent pom

Reply:
@stliu I know the spring poms are using the azure-client-sdk-parent and the starters do not. The changes in this PR should not get checked in for the other reasons I'd mentioned. It's not necessary and, on top of that, it's also not right. If the common code was something that was frequently changing we'd have to find a away around things it but that's not the case.